lib/fluent/log.rb

```diff
@@ -281,6 +284,7 @@ def event(level, args)
       }
 
       map.each_pair {|k,v|
+        next if (k.to_s == "plugin_type" || k.to_s == "plugin_id") and @optional_header
```

Comment:
No, this is not very exceptional case.

Currently, following cases are used `type` and `plugin_id` in Hash:

```diff
diff --git a/lib/fluent/agent.rb b/lib/fluent/agent.rb
index 4767533..c7846da 100644
--- a/lib/fluent/agent.rb
+++ b/lib/fluent/agent.rb
@@ -81,10 +81,10 @@ module Fluent
       @started_filters.map { |f|
         Thread.new do
           begin
-            log.info "shutting down filter#{@context.nil? ? '' : " in #{@context}"}", type: Plugin.lookup_type_from_class(f.class), plugin_id: f.plugin_id
+            log.info "shutting down filter#{@context.nil? ? '' : " in #{@context}"}"
             f.shutdown
           rescue => e
-            log.warn "unexpected error while shutting down filter plugins", plugin: f.class, plugin_id: f.plugin_id, error_class: e.class, error: e
+            log.warn "unexpected error while shutting down filter plugins", error_class: e.class, error: e
             log.warn_backtrace
           end
         end
@@ -95,10 +95,10 @@ module Fluent
       @started_outputs.map { |o|
         Thread.new do
           begin
-            log.info "shutting down output#{@context.nil? ? '' : " in #{@context}"}", type: Plugin.lookup_type_from_class(o.class), plugin_id: o.plugin_id
+            log.info "shutting down output#{@context.nil? ? '' : " in #{@context}"}"
             o.shutdown
           rescue => e
-            log.warn "unexpected error while shutting down output plugins", plugin: o.class, plugin_id: o.plugin_id, error_class: e.class, error: e
+            log.warn "unexpected error while shutting down output plugins", error_class: e.class, error: e
             log.warn_backtrace
           end
         end
diff --git a/lib/fluent/root_agent.rb b/lib/fluent/root_agent.rb
index 4814e20..624b661 100644
--- a/lib/fluent/root_agent.rb
+++ b/lib/fluent/root_agent.rb
@@ -123,7 +123,7 @@ module Fluent
       @started_inputs.map { |i|
         Thread.new do
           begin
-            log.info "shutting down input", type: Plugin.lookup_type_from_class(i.class), plugin_id: i.plugin_id
+            log.info "shutting down input"
             i.shutdown
           rescue => e
             log.warn "unexpected error while shutting down input plugin", plugin: i.class, plugin_id: i.plugin_id, error_class: e.class, error: e
```

Should we apply these changes?